lib/diagnostic.rb

```diff
@@ -15,28 +17,31 @@ def longest(list)
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.inject(0, :+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
-  end
+    if odds = numbers.select(&:odd?)
+      true
+    end
 
```

Comment:
1) I notice that you have a `def` and an `if` statement, but only 1 `end`.
2) what is line 27 doing? what is `odds`? and how can the if-statement evaluate an assignment expression? (recall that `=` is assignment, and `==` is checking equivalence)